Source/Core/Common/FileUtil.cpp

```diff
@@ -276,33 +279,68 @@ bool DeleteDir(const std::string& filename)
   return false;
 }
 
+// Repeatedly invokes func until it returns true or max_attempts failures.
+// Waits after each failure, with each delay doubling in length.
+template <typename FuncType>
+static bool AttemptMaxTimesWithExponentialDelay(const FuncType& func, int max_attempts,
+                                                std::chrono::milliseconds delay)
+{
+  for (int failed_attempts = 0; failed_attempts < max_attempts; ++failed_attempts)
+  {
+    if (func())
+    {
+      return true;
+    }
+    if (failed_attempts + 1 < max_attempts)
+    {
+      INFO_LOG_FMT(COMMON, "Rename attempt failed, delaying for {} milliseconds", delay.count());
+      std::this_thread::sleep_for(delay);
+      delay *= 2;
+    }
+  }
+  return false;
+}
+
 // renames file srcFilename to destFilename, returns true on success
 bool Rename(const std::string& srcFilename, const std::string& destFilename)
 {
   INFO_LOG_FMT(COMMON, "Rename: {} --> {}", srcFilename, destFilename);
 #ifdef _WIN32
-  auto sf = UTF8ToTStr(srcFilename);
-  auto df = UTF8ToTStr(destFilename);
-  // The Internet seems torn about whether ReplaceFile is atomic or not.
-  // Hopefully it's atomic enough...
-  if (ReplaceFile(df.c_str(), sf.c_str(), nullptr, REPLACEFILE_IGNORE_MERGE_ERRORS, nullptr,
-                  nullptr))
-    return true;
-  // Might have failed because the destination doesn't exist.
-  if (GetLastError() == ERROR_FILE_NOT_FOUND)
-  {
-    if (MoveFile(sf.c_str(), df.c_str()))
+  // Store converted strings in local variables so cstrings don't point to temporary objects
+  const std::wstring source_wstring = UTF8ToTStr(srcFilename);
+  const std::wstring destination_wstring = UTF8ToTStr(destFilename);
+  const wchar_t* const source_ptr = source_wstring.c_str();
+  const wchar_t* const destination_ptr = destination_wstring.c_str();
+
+  // On Windows ReplaceFile can fail spuriously due to antivirus checking or other noise.
+  // Retry the operation with increasing delays, and if none of them work there's probably a
+  // persistent problem.
+  const auto attempt_func = [&, source_ptr, destination_ptr]() {
+    if (ReplaceFile(destination_ptr, source_ptr, nullptr, REPLACEFILE_IGNORE_MERGE_ERRORS, nullptr,
+                    nullptr))
+    {
       return true;
-  }
-  ERROR_LOG_FMT(COMMON, "Rename: MoveFile failed on {} --> {}: {}", srcFilename, destFilename,
-                GetLastErrorString());
+    }
+    // Might have failed because the destination doesn't exist.
+    if (GetLastError() == ERROR_FILE_NOT_FOUND)
+    {
+      return MoveFile(source_ptr, destination_ptr) != 0;
+    }
+    return false;
+  };
+  const bool success =
+      AttemptMaxTimesWithExponentialDelay(attempt_func, 3, std::chrono::milliseconds(5));
```

Comment:
Since the lambda isn't reused, it would be clearer to just not assign it to a variable and pass it directly to this function call.